gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagerUtils.java

```diff
@@ -366,4 +396,63 @@ static List<String> getDistinctUniqueRequesters(String serializedRequesters) {
       throw new RuntimeException("Could not process requesters due to ", e);
     }
   }
+
+  public static void submitInitializationEventsAndSetStatus(Dag<JobExecutionPlan> dag, Optional<EventSubmitter> eventSubmitter) {
```

Comment:
given the entire body is skipped when absent (e.g. you don't even set status when not submitting), it may be clearer to phrase as:
```
submitAndSet(Dag<JEP> dag, EventSubmitter es)
```
for use as:
```
eventSubmitter.ifPresent(es -> submitAndSet(dag, es);
```

to promote that, let's rid ourselves of the antiquated guava `Optional` in favor of the java8 one